### PR TITLE
weights: pallet-staking & pallet-bags-list

### PR DESCRIPTION
**Pull request by emostov**
Update weights following https://github.com/paritytech/polkadot/pull/3413

### CODE REVIEW

**Comment by emostov** [comment]
after the other westend one, just need to run "benchmark runtime westend pallet_staking" and then i think this is done

File: runtime/kusama/src/weights/pallet_bags_list.rs [hunk 1]
```diff
@@ -1,65 +1,64 @@
...
-		(75_718_000 as Weight)
+		(65_491_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(7 as Weight))
 			.saturating_add(T::DbWeight::get().writes(5 as Weight))
 	}
 	// Storage: Staking Bonded (r:1 w:0)
 	// Storage: Staking Ledger (r:1 w:0)
-	// Storage: BagsList ListNodes (r:4 w:4)
-	// Storage: BagsList ListBags (r:1 w:1)
+	// Storage: BagsList ListNodes (r:3 w:3)
```
Comment (kianenigma) [inline runtime/kusama/src/weights/pallet_bags_list.rs:58]: hmm why did the storage changed compared to substrate weights?

Comment (emostov) [reply runtime/kusama/src/weights/pallet_bags_list.rs:58]: Its a different benchmark, the terminal nodes r/w bags 2 but nodes 1 less time. The diff is because the old stuff was just copy pasta to get things to compile

File: runtime/kusama/src/weights/pallet_staking.rs [hunk 2]
```diff
@@ -42,181 +44,368 @@ use sp_std::marker::PhantomData;
...
-			// Standard Error: 1_000
-			.saturating_add((61_000 as Weight).saturating_mul(l as Weight))
-			.saturating_add(T::DbWeight::get().reads(3 as Weight))
-			.saturating_add(T::DbWeight::get().writes(3 as Weight))
+	// Storage: Staking Ledger (r:1 w:1)
+	// Storage: Balances Locks (r:1 w:1)
+	// Storage: System Account (r:1 w:1)
+	// Storage: BagsList ListNodes (r:3 w:3)
+	// Storage: Staking Bonded (r:1 w:0)
+	// Storage: BagsList ListBags (r:2 w:2)
```
Comment (kianenigma) [inline runtime/kusama/src/weights/pallet_staking.rs:281]: would be nice to make a follow up issue to consider refund for these operations. All of this is the worst case now (which is totally intentional)

File: runtime/kusama/src/weights/pallet_staking.rs [hunk 3]
```diff
@@ -42,181 +44,368 @@ use sp_std::marker::PhantomData;
...
-	fn get_npos_voters(v: u32, n: u32, s: u32) -> Weight {
+	// Storage: Staking CounterForNominators (r:1 w:0)
+	// Storage: Staking CounterForValidators (r:1 w:0)
+	// Storage: Staking Validators (r:501 w:0)
+	// Storage: Staking Bonded (r:1500 w:0)
+	// Storage: Staking Ledger (r:1500 w:0)
+	// Storage: Staking SlashingSpans (r:21 w:0)
+	// Storage: Staking Nominators (r:1001 w:0)
+	// Storage: System BlockWeight (r:1 w:1)
+	fn get_npos_voters(v: u32, n: u32, s: u32, ) -> Weight {
```
Comment (kianenigma) [inline runtime/kusama/src/weights/pallet_staking.rs:364]: hmm how is this benchmark not reading the bags list?

**Review by kianenigma — APPROVED** [review]
TLDR is that the fees/weight for staking in kusama is not more for some transactions, polkadot stays the same, as expected. LGTM, except my last comments.

File: runtime/kusama/src/weights/pallet_staking.rs (hunk 3 above)
Comment (emostov) [reply runtime/kusama/src/weights/pallet_staking.rs:364]: just because right now we don't iter over the bags, but instead the nominators map - https://github.com/paritytech/substrate/blob/master/frame/staking/src/pallet/impls.rs#L685 & https://github.com/paritytech/polkadot/blob/e0a53cf5beb87dd21fd50ef1d9be3ac39a0d7488/runtime/common/src/elections.rs#L106-L108

File: runtime/kusama/src/weights/pallet_staking.rs (hunk 2 above)
Comment (emostov) [reply runtime/kusama/src/weights/pallet_staking.rs:281]: https://github.com/paritytech/substrate/issues/9826

**Comment by emostov** [comment]
bot merge

**Comment by emostov** [comment]
bot merge force

**Comment by emostov** [comment]
bot merge

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Review by hellbill — APPROVED** [review]
